### PR TITLE
issue/1693 - Affiliate area tab improvements 

### PR DESCRIPTION
**Pull request by ramiabraham**
- Introduces `affwp_affiliate_dashboard_tabs`, which gets all affiliate dashboard tabs, and returns a filterable array, `affwp_affiliate_dashboard_tabs`.
- Replaces hardcoded affiliate dashboard tab markup with a loop through `affwp_affiliate_dashboard_tabs`.
- #1693

### CODE REVIEW

**Comment by DrewAPicture** [comment]
Is this going to affect the Affiliate Area Tabs add-on at all?

File: templates/dashboard.php [hunk 1]
```diff
@@ -53,16 +53,24 @@
 		<ul id="affwp-affiliate-dashboard-tabs">
 			<?php
 
-			$tabs = affwp_affiliate_dashboard_tabs();
+			$tabs = affwp_get_affiliate_dashboard_tabs();
+
+			// If the Affiliate Area Tabs add-on is hiding this tab, unset it from the $tabs array.
+			if ( class_exists( 'AffiliateWP_Affiliate_Area_Tabs' ) ) {
```
Comment (pippinsplugins) [inline templates/dashboard.php]: I don't believe this should be necessary. Affiliate Area Tabs uses the `affwp_affiliate_area_show_tab` filter to hide/show tabs. Am I wrong?

Comment (ramiabraham) [reply templates/dashboard.php]: It uses its own setting, `affiliate_area_hide_tabs`, which is checked here. If an affiliate area page is found, it's `unset` from the array.

Comment (pippinsplugins) [reply templates/dashboard.php]: @ramiabraham can we adjust it so the logic is purely in the add-on? As a general rule, we try to not put any add-on-specific code in core. The add-ons should extend and modify core, not the other way around.

**Comment by ramiabraham** [comment]
@amdrew @pippinsplugins 
With AAT compatibility removed in https://github.com/AffiliateWP/AffiliateWP/pull/2386/commits/d53d2fa81bb85fb4856a99c5142cdc65e408401b, [this issue](https://github.com/AffiliateWP/affiliatewp-affiliate-area-tabs/issues/47) in AAT would need to be resolved in AAT prior to the 2.1.6 release (asking for feedback in review above).

**Comment by pippinsplugins** [comment]
@ramiabraham looks like we have a merge conflict now.

File: includes/affiliate-functions.php [hunk 2]
```diff
@@ -1506,6 +1506,120 @@ function affwp_get_active_affiliate_area_tab() {
...
+		}
+	}
+
+	/**
+	 * Filters the affiliate dashboard tabs.
+	 * Used to add, remove, or modify affiliate dashboard tabs.
+	 *
+	 * To add a new tab, provide an array containing the following:
+	 *
+	 *   $tabs['unique_tab_id'] => array(
```
Comment (amdrew) [inline includes/affiliate-functions.php]: @ramiabraham Can we change `=>` to `=`

Comment (ramiabraham) [reply includes/affiliate-functions.php]: Updated!

File: includes/affiliate-functions.php [hunk 3]
```diff
@@ -1474,35 +1472,235 @@ function affwp_get_affiliate_area_page_url( $tab = '' ) {
...
+	/**
+	 * Filters the affiliate dashboard tabs.
+	 * Used to add, remove, or modify affiliate dashboard tabs.
+	 *
+	 * To add a new tab, provide an array containing the following:
+	 *
+	 *   $tabs['id'] = array(
+	 *      'title'    => __( 'Settings', 'affiliate-wp' ),
+	 *		'content'  => '',
+	 *		'priority' => 10
```
Comment (amdrew) [inline includes/affiliate-functions.php]: Indentation on content and priority needs tweaking

**Pull request closed without merging**